toolkit/scripts/test_package_install.sh

```diff
@@ -0,0 +1,199 @@
+# -s -> packages to test for install and uninstall functionality
+# -i -> packages to not install
+# -u -> packages to not uninstall
+# -y -> daily build ID
+# -d -> dist tag
+# -v -> verbose output
+
+while getopts ":p:s:i:u:y:d:v" OPTIONS; do
+  case "${OPTIONS}" in
+    p ) RPM_DIRECTORY=$OPTARG;;
+    s ) RPMS_TO_INSTALL=${OPTARG//,/ } ;;
+    i ) PACKAGES_TO_NOT_INSTALL=${OPTARG//,/ } ;;
+    u ) PACKAGES_TO_NOT_UNINSTALL=${OPTARG//,/ } ;;
+    y ) DAILY_BUILD_ID=$OPTARG ;;
+    d ) DIST_TAG=$OPTARG ;;
+    v ) set -x ;;
+
+    \? )
+        echo "Error - Invalid Option: -$OPTARG" 1>&2
+        exit 1
+        ;;
+    : )
+        echo "Error - Invalid Option: -$OPTARG requires an argument" 1>&2
+        exit 1
+        ;;
+  esac
+done
+
+ARCHITECTURE=$(uname -p)
+
+# Determine LKG_BASE_URL
+if [[ -n $DAILY_BUILD_ID &&
+      $DAILY_BUILD_ID == "latest" &&
+      $DIST_TAG == ".azl3" ]]; then
+    echo "Determine LKG_BASE_URL directly from lkg-3.0-dev.json"
+    wget -nv https://mariner3dailydevrepo.blob.core.windows.net/lkg/lkg-3.0-dev.json
+    LKG_BASE_URL=$(cat lkg-3.0-dev.json | jq -r .repo.$ARCHITECTURE)
+else
+    echo "Determine LKG_BASE_URL from provided DAILY_BUILD_ID"
+    if [[ $ARCHITECTURE == "x86_64" ]]; then
+        DAILY_BUILD_ARCH="-x86-64"
+    else
+        DAILY_BUILD_ARCH="-aarch64"
+    fi
+    LKG_BASE_URL=https://mariner3dailydevrepo.blob.core.windows.net/daily-repo-$DAILY_BUILD_ID$DAILY_BUILD_ARCH
+fi
+
+echo "---------------------------------------------------------------"
+echo "-- Install packages in provided folder --"
+echo "---------------------------------------------------------------"
+echo "-- RPM_DIRECTORY                            -> $RPM_DIRECTORY"
+echo "-- RPMS_TO_INSTALL                          -> $RPMS_TO_INSTALL"
+echo "-- LKG_BASE_URL                             -> '$LKG_BASE_URL'"
+echo "-- DAILY_BUILD_ID                           -> '$DAILY_BUILD_ID'"
+echo "-- DAILY_BUILD_ARCH                         -> '$DAILY_BUILD_ARCH'"
+echo "-- DIST_TAG                                 -> '$DIST_TAG'"
+echo "-- PACKAGES_TO_NOT_INSTALL                  -> $PACKAGES_TO_NOT_INSTALL"
+echo "-- PACKAGES_TO_NOT_UNINSTALL                -> $PACKAGES_TO_NOT_UNINSTALL"
+echo ""
+
+echo "##[debug]Installing these space separated packages:"
+echo "$RPMS_TO_INSTALL"
+echo ""
+
+RPMS_TO_INSTALL=($RPMS_TO_INSTALL)
+PACKAGES_TO_NOT_INSTALL=($PACKAGES_TO_NOT_INSTALL)
+PACKAGES_TO_NOT_UNINSTALL=($PACKAGES_TO_NOT_UNINSTALL)
+
+
+exists_in_list() {
+    local VALUE=$1
+    shift
+    local LIST=("$@")
+
+    for x in "${LIST[@]}"; do
+        if [ "$x" = "$VALUE" ]; then
+            return 0
+        fi
+    done
+    return 1
+}
+
+# Add built packages as a local repo for tdnf
+sudo createrepo --compatibility "$RPM_DIRECTORY"
+
+repo_content=$(cat << EOF
+[builtpackages]
+name=Local Built Packages($ARCHITECTURE)
+baseurl=file://$RPM_DIRECTORY
+enabled=1
+priority=0
+skip_if_unavailable=True
```

Comment:
Shouldn't it be `false`? Isn't there a possibility of the package being pulled from the remote?